lib/charms/opensearch/v0/opensearch_plugin_manager.py

```diff
@@ -311,32 +370,28 @@ def _user_requested_to_enable(self, plugin: OpenSearchPlugin) -> bool:
     def _is_enabled(self, plugin: OpenSearchPlugin) -> bool:
         """Returns true if plugin is enabled.
 
-        The main question to answer is if we would remove the configuration from
-        opensearch.yaml and/or add some other configuration instead.
-        If yes, then we know that the service is enabled.
-
-        The disable() method is used, as a given entry will be removed from opensearch.yml.
+        The main question to answer is if we would have it in the configuration
+        from cluster settings. If yes, then we know that the service is enabled.
 
-        If disable() returns a list, then check if the keys in the stored_plugin_conf
-        matches the list values; otherwise, if a dict, then check if the keys and values match.
-        If they match in any of the cases above, then return True.
+        Check if the configuration from enable() is present or not.
         """
         try:
-            plugin_conf = plugin.disable().config_entries_to_del
-            keystore_conf = plugin.disable().secret_entries_to_del
-            stored_plugin_conf = self._opensearch_config.get_plugin(plugin_conf)
-
-            if any(k not in self._keystore.list() for k in keystore_conf):
+            current_settings, new_conf = self._compute_settings(plugin.config())
+            if current_settings != new_conf:
                 return False
 
-            # Using sets to guarantee matches; stored_plugin_conf will be a dict
-            if isinstance(plugin_conf, list):
-                return set(plugin_conf) == set(stored_plugin_conf)
-            else:  # plugin_conf is a dict
-                return plugin_conf == stored_plugin_conf
+            # Now, focus in on the keystore part
+            keys_available = self._keystore.list()
```

Comment:
Does adding/remove a key warrant a restart?

Reply:
@Mehdi-Bendriss no, only writing to the config files, why? This is about deciding the current plugin status and if we have the keys written to the keystore

Reply:
That logic is to decide if a plugin is enabled, i.e. (1) it exists; and (2) the config options that enable it are present. That also includes the keys in the keystore. This method does not decide if we need a restart or not.